adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -275,7 +275,17 @@ private AuthenticationResult tryAcquireTokenSilent(final AuthenticationRequest a
 
         if (shouldTrySilentFlow(authenticationRequest)) {
             Logger.v(TAG, "Try to acquire token silently, return valid AT or use RT in the cache.");
-            authenticationResult = acquireTokenSilentFlow(authenticationRequest);
+            try {
```

Comment:
The point of this fix is that when using the binding service, we did not check if it is a silent call before throwing AuthenticationException(ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED).